operator/controllers/operation_controller_test.go

```diff
@@ -0,0 +1,299 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"reflect"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+
+	operatorv1 "k8s.io/kubeadm/operator/api/v1alpha1"
+)
+
+func TestOperatorReconciler_reconcilePhase(t *testing.T) {
+	tx := metav1.Now()
+	errMessage := "error"
+
+	type args struct {
+		operation *operatorv1.Operation
+	}
+	type want struct {
+		operation *operatorv1.Operation
+	}
+	tests := []struct {
+		name string
+		args args
+		want want
```

Comment:
can this be?
```
name string
input *operatorv1.Operation
output *operatorv1.Operation
```

or both `input` and `output` using the same type instead of separate `args`, `want`?